src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_P256.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      HSM based implementation of CHIP crypto primitives
+ *      Based on configurations in CHIPCryptoPALHsm_config.h file,
+ *      chip crypto apis use either HSM or rollback to software implementation.
+ */
+
+#include "CHIPCryptoPALHsm_SE05X_utils.h"
+
+#if ENABLE_HSM_GENERATE_EC_KEY
+
+#define MAX_SHA_ONE_SHOT_DATA_LEN 900
+#define NIST256_HEADER_OFFSET 26
+
+namespace chip {
+namespace Crypto {
+
+P256KeypairHSM::~P256KeypairHSM()
+{
+    if (keyid != 0)
+    {
+        if (provisioned_key == false)
+        {
+            ChipLogDetail(Crypto, "Deleting key with id - %x !", keyid);
+            se05x_delete_key(keyid);
+        }
+        else
+        {
+            ChipLogDetail(Crypto, "Provisioned key ! Not deleting key in HSM");
+        }
+    }
+}
+
+CHIP_ERROR P256KeypairHSM::Initialize()
+{
+    CHIP_ERROR error       = CHIP_ERROR_INTERNAL;
+    sss_status_t status    = kStatus_SSS_Success;
+    sss_object_t keyObject = { 0 };
+    uint8_t pubkey[128]    = {
+        0,
+    };
+    size_t pubKeyLen   = sizeof(pubkey);
+    size_t pbKeyBitLen = sizeof(pubkey) * 8;
+
+    if (keyid == 0)
+    {
+        ChipLogDetail(Crypto, "Keyid not set !. Set key id using 'SetKeyId' member class !");
+        goto exit;
```

Comment:
nit: probably ExitNow would be more consistent with existin macro style.

Since there is no cleanup happening in the exit block, return macros may be easier to read.

Reply:
Added ExitNow